lib/SimpleSAML/Metadata/Sources/MDQ.php

```diff
@@ -309,6 +320,12 @@ public function getMetaData(string $entityId, string $set): ?array
         $entity = SAMLParser::parseString($xmldata);
         Logger::debug(sprintf('%s: completed parsing of [%s]', __CLASS__, $mdq_url));
 
+        if (!empty($this->validateCertificate)) {
+            if (!$entity->validateSignature($this->validateCertificate)) {
+                throw new Exception(__CLASS__ . ': error, could not verify signature for entity: ' . $entityId . '".');
+            }
+        }
```

Comment:
Should we log that we're not validating anything in an `else` clause? At the very least for debugging, but maybe even as high as `notice` makes sense?

At least the current situation does not comply with your stated goal that verification must be done always.

Reply:
I've been discussing this with Jaime last night.. He feels like we shouldn't do _any_ signature verification here because the integrity is already guaranteed at TLS-level..  I am no longer sure what to do here.

Reply:
I think the current code guarantees nowhere that the configured URL is TLS.

It's fine to keep it optional with me but I think we'd need to have the signature verification just like we have on other forms of metadata (because regular metadata is also routinely served over TLS). Would in any case add the else-clause with logging though, even if only at info level. Just so you can follow what it's doing.